var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -189,18 +195,17 @@ class Llvm(CMakePackage, CudaPackage):
     # https://bugs.llvm.org/show_bug.cgi?id=39696
     patch("thread-p9.patch", when="@develop+libcxx")
 
-    @run_before("cmake")
-    def check_darwin_lldb_codesign_requirement(self):
-        if not self.spec.satisfies("+lldb platform=darwin"):
-            return
-        codesign = which("codesign")
-        mkdir("tmp")
-        llvm_check_file = join_path("tmp", "llvm_check")
-        copy("/usr/bin/false", llvm_check_file)
-
+    @run_before('cmake')
+    def codesign_check(self):
+        if '+code_signing' in self.spec:
+            codesign = which('codesign')
+            mkdir('tmp')
+            llvm_check_file = join_path('tmp', 'llvm_check')
+            copy('/usr/bin/false', llvm_check_file)
         try:
```

Comment:
You're going to need to indent this, as `codesign` won't be defined in the else case.

Reply:
You can also invert it and return if it's not needed like we used to do